### PR TITLE
[Dynamo] Fix bug of torch.is_floating_point & is_complex

### LINKED ISSUE #95192: Pytorch 2.0 [compile] is_floating and is_complex causes list index out of range
### 🐛 Describe the bug

Please use below code to reproduce failures:
```
import torch
class BasicModule(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, tensor):
        return torch.is_floating(input=tensor)

if __name__ == "__main__":
    tensor = torch.randn([1024], dtype=torch.float32)
    model = BasicModule()
    model = torch.compile(model)
    fwd_res = model(tensor)
    print(fwd_res)
```

Error:
```
File "/home/jthakur/.pt_2_0/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 338, in wrapper
    return inner_fn(self, inst)
  File "/home/jthakur/.pt_2_0/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 1007, in CALL_FUNCTION_KW
    self.call_function(fn, args, kwargs)
  File "/home/jthakur/.pt_2_0/lib/python3.8/site-packages/torch/_dynamo/symbolic_convert.py", line 466, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/jthakur/.pt_2_0/lib/python3.8/site-packages/torch/_dynamo/variables/torch.py", line 236, in call_function
    and isinstance(args[0], TensorVariable)
IndexError: list index out of range
```

Same code works fine, when run without compile 

### Versions

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==2.0.0.dev20230209+cpu
[pip3] torchaudio==2.0.0.dev20230209+cpu
[pip3] torchvision==0.15.0.dev20230209+cpu

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #95192

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/variables/torch.py [hunk 1]
```diff
@@ -234,21 +234,25 @@ def call_function(
...
+            torch.is_floating_point,
+            torch.is_complex,
         ):
-            if self.value is torch.is_floating_point:
-                return ConstantVariable(args[0].dtype.is_floating_point, **options)
-            elif self.value is torch.is_complex:
-                return ConstantVariable(args[0].dtype.is_complex, **options)
+            input_arg = None
+            if args:
+                input_arg = args[0]
```
Comment (ezyang) [inline torch/_dynamo/variables/torch.py:243]: Don't we have to test that there are not too many arguments? (This seems to be a preexisting condition)

File: torch/_dynamo/variables/torch.py [hunk 2]
```diff
@@ -234,21 +234,25 @@ def call_function(
...
+                input_arg = kwargs["input"]
+            if isinstance(input_arg, TensorVariable) and input_arg.dtype is not None:
+                if self.value is torch.is_floating_point:
+                    return ConstantVariable(
+                        input_arg.dtype.is_floating_point, **options
+                    )
+                elif self.value is torch.is_complex:
+                    return ConstantVariable(input_arg.dtype.is_complex, **options)
+                else:
+                    raise AssertionError()
```
Comment (ezyang) [inline torch/_dynamo/variables/torch.py:255]: I'd recommend printing self.value in the assert message

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
While I agree the reimplementation is correct, it is annoying and weird that the regular ops path does not work: torch.is_floating_point is a dispatched operator and you should be able to just pass the fake tensor to it directly without any special casing.

File: torch/_dynamo/variables/torch.py (hunk 1 above)
Comment (yanboliang) [reply torch/_dynamo/variables/torch.py:243]: It's good to have, but usually we don't have unit tests for scenario that even don't pass eager mode.

**Comment by voznesenskym** [comment]
> While I agree the reimplementation is correct, it is annoying and weird that the regular ops path does not work: torch.is_floating_point is a dispatched operator and you should be able to just pass the fake tensor to it directly without any special casing.

Have we tried this? This code could just be doing too much from pre-fake days?

**Pull request closed without merging**